enhancements/sig-architecture/123-addon-multiple-configs/README.md

```diff
@@ -0,0 +1,230 @@
+	// +optional
+	Configs []AddOnConfig `json:"configs,omitempty"`
+```
+
+### Effective addon configs
+
+As the motivation mentioned, today there are 3 places to set addon configurations. 
+
+Currently, since each GVK supports only one config, the GVK is used as the identifier for configs. When configs with same GVK configured in multiple places, it follows the override sequence and 
+only one config is listed as the effective config in the mca.status.
+
+With the support for multiple configs with the same GVK, `GVK + namespace + name` becomes the identifier. 
+We need to reconsider the logic for setting mca.status when configuring multiple configs in multiple places.
+
+Using the example below (omitting the group and namespace for simplicity), we present three options. 
+We do not expect to support all, but want to involve discussion and finally only choose one.
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: example
+spec:
+  supportedConfigs:
+    - resource: r1
+      defaultConfig:
+        name: n1
+    - resource: r2
+  installStrategy:
+    type: Placements
+    placements:
+      - name: example
+        namespace: example
+        configs:
+          - resource: r2
+            name: n1
+          - resource: r2
+            name: n2
+
+
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddOn
+metadata:
+  name: example
+  namespace: cluster1
+spec:
+  installNamespace: open-cluster-management-agent-addon
+  configs:
+    - resource: r2
+      name: n3
+```
+
+#### Merge
+
+With `GVK + namespace + name` as the identifier for each config, there is no override concept now actually, all the configured configs will be listed in the mca.status. 
+
+Configs will be ordered, with the last configured on top. For example, configs in mca will be on top of cma configs, and mca configs with higher indices 
+will be on top of those with lower indices.
+
+The addon's own controller will determine how to use the configs (e.g., choose one, read all, etc).
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddOn
+metadata:
+  name: example
+  namespace: cluster1
+status:
+...
+  configReferences:
+  - desiredConfig:
+      name: n3
+      specHash: <spec-hash>
+    name: n3
+    resource: r2
+  - desiredConfig:
+      name: n2
+      specHash: <spec-hash>
+    name: n2
+    resource: r2
+  - desiredConfig:
+      name: n1
+      specHash: <spec-hash>
+    name: n1
+    resource: r2
+  - desiredConfig:
+      name: n1
+      specHash: <spec-hash>
+    name: n1
+    resource: r1
+```
+
+#### Override by GVK
+
+"Override by GVK" is a way try to keep consistent with current behavior. 
+
+The gaps is if the r2n1 and r2n2 resource are actually some global configs, you may not want it to be overriden by r2n3. Compared with "Merge" options, 
+this options can not cover such case and is lack of flexibility.
```

Comment:
I see this point but at the same time, this approach is more similar to the current behaviour. However, "global configs" use case is a valid one that maybe could be addressed in a follow-up enhancement, I was thinking along the lines of CMA having a dedicated place for default configs like it does for `defaultConfigs` (just an idea).